Migrate old Quick Start UI code to ImprovedMySiteFragment as Quick Start Block

ashiagr opened the pull request:
Issue #15127

This PR migrates old `Quick Start` UI code from `MySiteFragment` to `ImprovedMySiteFragment` as a `Quick Start Block`.

To test:

1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Disable `QuickStartDynamicCardsFeatureConfig`, enable `MySiteImprovementsFeatureConfig`
3. Restart the app
4. Select a site with quick start in progress 
5. Notice that the old quick start section gets displayed according to completed/ uncompleted steps

![qs_improved_my_site](https://user-images.githubusercontent.com/1405144/128358824-8cf0b450-dfdf-482e-8097-be2b14607de3.png)

## Regression Notes
1. Potential unintended areas of impact
    - `ImprovedMySiteFragment` -> `Quick Start - Dynamic Card` (`MySiteImprovementsFeatureConfig` - ON, `QuickStartDynamicCardsFeatureConfig` - ON )
    - `MySiteFragment` -> `Quick Start` (`MySiteImprovementsFeatureConfig` - OFF)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for `Quick Start Block` display.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr committed:
Rename quick start dynamic card type

ashiagr committed:
Add empty quick start block

ashiagr committed:
Add empty quick start block view holder

Quick start layout content copied as it is from my_site_fragment

ashiagr committed:
Display quick start block content

Content is straightaway copied from MySiteFragment -> updateQuickStartContainer(). It will be refactored to get from a builder class in a later commit.

ashiagr committed:
Get quick start card content using builder

ashiagr committed:
Move isQuickStartInProgress condition to VM

ashiagr committed:
Tests: Add quick start block VM tests

ashiagr committed:
Rename isCompleted to strikeThroughTitle based on usage

ashiagr committed:
Tests: Add tests for strikeThroughTitle

ashiagr committed:
Tests: Add quick start block builder tests

ashiagr committed:
Use correct siteId when checking quick start progress

MySiteFragment -> updateQuickStartContainer uses site id saved in the AppPrefs. It doesn't seem to be the always same as the one from the selected site repository.

ashiagr committed:
Merge branch 'develop' into issue/15126-migrate-old-qs-ui-to-improved-my-site

zwarm commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt:
```diff
@@ -252,7 +254,7 @@ class MySiteViewModel
 
             if (!quickStartDynamicCardsFeatureConfig.isEnabled() &&
-                    quickStartUtilsWrapper.isQuickStartInProgress(site.siteId.toInt())) {
+                    quickStartUtilsWrapper.isQuickStartInProgress(appPrefsWrapper.getSelectedSite())) {
```
@ashiagr - DynamicCards source uses SiteRepository for siteId which monitors the site selection change.  I wonder why MySiteFragment uses AppPrefs?  🤔  Makes me question if siteId was wrong for those 1 and 1/2 months that ImprovedMySiteFragment was out OR maybe a matter of everything so intertwined.

ashiagr replied:
Yep, it is so much intertwined 🙂. As discussed, added it as an investigation task in the [parent issue](#15127).

zwarm reviewed (commented):
@ashiagr  - great start and it was easy to review since you walked me through most of it this morning. Nice separation into the builder and +1 on the elimination of the store from the adapter. The "siteId" use is my only question... I called it out separately.

ashiagr committed:
Merge branch 'develop' into issue/15126-migrate-old-qs-ui-to-improved-my-site

ashiagr committed:
Fix lint

To set quick start task type ite subtitle in the builder class, UiStringWithParams requires string arguments.
To fix the lint error, created another string quick_start_sites_type_tasks_completed and reverted quick_start_sites_type_subtitle used in MySiteFragment to have int arguments.

zwarm reviewed (approved):
All good, thanks for taking care of Lint. Yes, siteId is a much larger discussion. Thanks!!
